safenode/src/network/mod.rs

```diff
@@ -258,27 +265,28 @@ impl Network {
     }
 
     /// Find the closest peers to the given `XorName`
-    pub async fn get_closest_peers(&self, xor_name: XorName) -> Result<HashSet<PeerId>> {
+    /// If `client` is false, then include `self` among the `closest_peers`
+    pub async fn get_closest_peers(&self, xor_name: XorName, client: bool) -> Result<Vec<PeerId>> {
```

Comment:
I think it will be clearer to have two pub functions then both call a priv fn with different flag, so lookes like:
```rust
pub async fn client_get_closest_peers(&self, xor_name: XorName) -> Result<Vec<PeerId>> {
    self.get_closest_peers(xor_name, true).await
}
pub async fn node_get_closest_peers(&self, xor_name: XorName) -> Result<Vec<PeerId>> {
    self.get_closest_peers(xor_name, false).await
}
async fn get_closest_peers(&self, xor_name: XorName, is_client: bool) -> Result<Vec<PeerId>> {
    ......
}
```